nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -131,9 +135,19 @@
     @WritesAttribute(attribute = "invokehttp.java.exception.message", description = "The Java exception message raised when the processor fails"),
     @WritesAttribute(attribute = "user-defined", description = "If the 'Put Response Body In Attribute' property is set then whatever it is set to "
         + "will become the attribute key and the value would be the body of the HTTP response.")})
-@DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
-                    description = "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
-                            + "the Attribute Expression Language set in the value of the Dynamic Property.")
+@DynamicProperties ({
+    @DynamicProperty(name = "Header Name", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "Send request header with a key matching the Dynamic Property Key and a value created by evaluating "
+                + "the Attribute Expression Language set in the value of the Dynamic Property."),
+    @DynamicProperty(name = "post.form.<NAME>", value = "Attribute Expression Language", expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES,
+        description =
+            "When the HTTP Method is POST, dynamic properties with the property name in the form of post.form.<NAME>,"
+                + " where the <NAME> will be the form data name, will be used to fill out the multipart form parts."
+                + "  If the value is the literal 'FLOWFILE_CONTENT', that form part will have the flowfile's data."
+                + "  The property name post.form.filename will set the filename to set for the flowfile content."
+                + "  If send message body is false, the properties will be ignored.")
```

Comment:
Which properties will be ignored? All post.form.<name> properties or just the one with a value of FLOWFILE_CONTENT and the post.form.filename?

Reply:
All post.form properties are ignored if send message body is false. That is more in line with how the current code worked.  I can see a logical argument that if there are forms set, you should send those and only let the send body apply to sending the flowfile.  What do you think?

Reply:
Actually, the more I think of it, the possibility of just sending form data and no file is real.  This should be changed.  Although I don't know if that wrecks the send body idea. Thoughts?

Reply:
I think it makes perfect sense to allow sending of form parameters but not the FlowFile content.

Reply:
This is done.  They can send just forms, just the flow file _as a form_, or the flowfile